src/schemas/chat-completion.ts

```diff
@@ -13,25 +15,60 @@ export const ChatMessageRoleSchema = z.union([
   z.literal('system'),
   z.literal('user'),
   z.literal('assistant'),
+  z.literal('function'),
 ]);
 export type ChatMessageRole = z.infer<typeof ChatMessageRoleSchema>;
 
+/** Chat completion function call */
+export const ChatMessageFunctionCallSchema = z.object({
+  /** The name of the function to call. */
+  name: z.string(),
+  /** The arguments to call the function with, as generated by the model in JSON format. Note that the model does not always generate valid JSON, and may hallucinate parameters not defined by your function schema. Validate the arguments in your code before calling your function. */
+  arguments: z.string(),
```

Comment:
The [node library types](https://github.com/openai/openai-node/commit/b7cbca5852b4460e283c2cbed611d641e95c9e81#diff-216c63c6346d7d78fcd3b1ceeae8c9772b30f8fb2d43b8beabcda6ebd6aa6660R103) has `name` and `arguments` values as optional and the [API reference](https://platform.openai.com/docs/api-reference/chat/create#chat/create-function_call) docs don't specify 🤔

Reply:
Yeah; I think they may have multiple `FunctionCall` objects, since in both of these examples, it's the JSON stringified version of the `arguments` instead of the JSON version.

<img width="777" alt="CleanShot 2023-06-13 at 20 47 53@2x" src="https://github.com/rileytomasek/openai-fetch/assets/552829/f1c2ca75-cfaa-4787-9b07-aaa5d369b309">

<img width="901" alt="CleanShot 2023-06-13 at 20 49 40@2x" src="https://github.com/rileytomasek/openai-fetch/assets/552829/734450b2-5891-49a2-96f8-b87574b3742e">

https://openai.com/blog/function-calling-and-other-api-updates

Reply:
Hmmm I'm not sure; I guess we should make them optional to guarantee that `zod` doesn't complain?